dialogues/classify_water_dialog.py

```diff
@@ -0,0 +1,421 @@
+
+            full_img_array = envi_img.load()
+            img_array = full_img_array[:,:,(rgb)]
+            img_array = ((img_array - img_array.min()) / img_array.max() * 255).astype(np.uint8)
+            np_array = np.array(img_array)
+
+            img = cv2.cvtColor(np_array, cv2.COLOR_BGR2RGB)
+
+            return img, full_img_array, envi_img
+        
+        except Exception as err:
+            print("Error when getting image from hdr: " + str(err))
+
+
+    def show_img(self, img, label, paint=False):
+        height, width, _ = img.shape
+        bytes_per_line = 3 * width
+        q_img = QImage(img.data, width, height, bytes_per_line, QImage.Format_RGB888).rgbSwapped()
+
+        if paint:
+            self.paint_pixels(q_img)
+
+        label.setPixmap(QPixmap.fromImage(q_img))
+
+
+    def paint_pixels(self, q_img):
+        painter = QPainter(q_img)
+        painter.setPen(QPen(QColor(255, 0, 0), 10))
+        painter.drawPoint(self.water_pixel_pos_1[0], self.water_pixel_pos_1[1])
+
+        painter.setPen(QPen(QColor(0, 255, 0), 10))
+        painter.drawPoint(self.water_pixel_pos_2[0], self.water_pixel_pos_2[1]) 
+        
+        painter.setPen(QPen(QColor(255, 255, 0), 10))
+        painter.drawPoint(self.water_pixel_pos_3[0], self.water_pixel_pos_3[1]) 
+        painter.end()
+
+
+    def set_exp(self):
+        if self.input_exp.text() != "":
+            self.exp = float(self.input_exp.text())
+
+
+    def perform_cube_calibration(self):
+        try:
+            bip_path = self.filepath[:-4] + ".bip"
+            cube_calibrated, self.calibrated_wl, self.cube_scaled = cube_calibration(bip_path, self.exp)
+
+            # Cretes a new calibrated .bip and .hdr file in the same directory
+            cube_calibrated.astype(dtype = "uint16").tofile(self.filepath[:-4] + "_calibrated.bip")
+            calibrated_hdr_path = shutil.copy2(self.filepath, self.filepath[:-4] + "_calibrated.hdr")   
+
+            self.calibrated_img, self.calibrated_img_array, self.calibrated_envi_img = self.get_img_from_hdr(calibrated_hdr_path)
+            self.calibrated_img = self.tune_overexposed_img(self.cube_scaled[:,::-1,:], cube_calibrated[:,::-1,:], wl=self.calibrated_wl)
+            self.show_img(self.calibrated_img, self.label_new_img, paint=True)
+
+            self.new_img_status  = "calibrated"
+
+        except Exception as err:
+            print("Error when performing cube calibration: " + str(err))
+
+
+    def set_pixels(self):
+        if self.input_pixel_1.text() != "":
+            self.water_pixel_pos_1 = tuple(map(int, self.input_pixel_1.text().split(",")))
+        if self.input_pixel_2.text() != "":
+            self.water_pixel_pos_2 = tuple(map(int, self.input_pixel_2.text().split(",")))
+        if self.input_pixel_3.text() != "":
+            self.water_pixel_pos_3 = tuple(map(int, self.input_pixel_3.text().split(",")))
+        
+        self.perform_cube_calibration()
```

Comment:
Function "set_pixels()" performs cube calibration? self.perform_cube_calibration() should be moved (a function should do one thing)

Reply:
fixed